cosmoz-omnitable.js

```diff
@@ -275,65 +311,109 @@
 			});
-			this.$.groupedList.scrollTarget = this.$.scroller;
-			this.listen(this, 'cosmoz-column-hidden-changed', '_debounceUpdateColumns');
-			this.listen(this, 'cosmoz-column-disabled-changed', '_debounceUpdateColumns');
 
+			this.$.groupedList.scrollTarget = this.$.scroller;
+			this.addEventListener('cosmoz-column-hidden-changed', this._debounceUpdateColumns);
+			this.addEventListener('cosmoz-column-disabled-changed', this._debounceUpdateColumns);
+
+			this.addEventListener('iron-resize', this._onResize);
+			this.addEventListener('update-item-size', this._onUpdateItemSize);
+			this.addEventListener('cosmoz-column-title-changed', this._onColumnTitleChanged);
+			this.addEventListener('cosmoz-column-filter-changed', this._filterChanged);
+			this.addEventListener('cosmoz-column-editable-changed', this._onColumnEditableChanged);
+			this.addEventListener('cosmoz-column-values-update', this._onColumnValuesUpdate);
 			this._fitDropdowns();
-		},
+		}
 
-		detached() {
+		disconnectedCallback() {
+			super.disconnectedCallback();
 			if (this._columnObserver) {
-				Polymer.dom(this).unobserveNodes(this._columnObserver);
+				this._columnObserver.disconnect();
+				this._columnObserver = null;
 			}
-			this.unlisten(this, 'cosmoz-column-hidden-changed', '_debounceUpdateColumns');
-			this.unlisten(this, 'cosmoz-column-disabled-changed', '_debounceUpdateColumns');
+
+			this.removeEventListener('cosmoz-column-hidden-changed', this._debounceUpdateColumns);
+			this.removeEventListener('cosmoz-column-disabled-changed', this._debounceUpdateColumns);
 			// Just in case we get detached before a planned debouncer has not run yet.
-			this.cancelDebouncer('adjustColumns');
-			this.cancelDebouncer('updateColumns');
-			this.cancelDebouncer('filterItems');
-			this.cancelDebouncer('groupItems');
-			this.cancelDebouncer('sortItems');
-		},
+			this._cancelDebouncers();
+
+			this.removeEventListener('iron-resize', this._onResize);
+			this.removeEventListener('update-item-size', this._onUpdateItemSize);
+			this.removeEventListener('cosmoz-column-title-changed', this._onColumnTitleChanged);
+			this.removeEventListener('cosmoz-column-filter-changed', this._filterChanged);
+			this.removeEventListener('cosmoz-column-editable-changed', this._onColumnEditableChanged);
+			this.removeEventListener('cosmoz-column-values-update', this._onColumnValuesUpdate);
+		}
 
-		/** ELEMENT BEHAVIOR */
+		flush() {
+			if (this._adjustColumnsDebouncer) {
+				this._adjustColumnsDebouncer.flush();
+			}
 
-		_disabledColumnsIndexes: null,
+			if (this._updateColumnsDebouncer) {
+				this._updateColumnsDebouncer.flush();
+			}
+
+			if (this._filterItemsDebouncer) {
+				this._filterItemsDebouncer.flush();
+			}
+
+			if (this._groupItemsDebouncer) {
+				this._groupItemsDebouncer.flush();
+			}
+
+			if (this._sortItemsDebouncer) {
+				this._sortItemsDebouncer.flush();
+			}
+		}
+
+		_cancelDebouncers() {
+			[this._adjustColumnsDebouncer,
```

Comment:
Not sure how to make this look nice and readable.. 

```suggestion
			[
				this._adjustColumnsDebouncer,
				this._updateColumnsDebouncer,
				this._filterItemsDebouncer,
				this._groupItemsDebouncer,
				this._sortItemsDebouncer
			]
				.filter(d => d != null)
				.forEach(d => d.cancel());
```

maybe?

`d != null` is both more expressive and faster (truthy/falsy statements is about 8 checks vs `!= null` at about 2)